### DISCUSSION TITLE
Delay in executing Http tasks when there are multiple http tasks in one workflow

### DISCUSSION THREAD

**Discussion by muralicmr** [Netflix/conductor#2992] [Q&A]
Hi,

We have a scenario wherein the workflow composes of multiple system tasks (HTTP, jsonjqt and fork-join). We observe a delay between two tasks being executed when we have a system running with concurrent requests such as 5 threads.  This can be seen in image below (delay between tasks are highlighted with yellow line). 

<img width="551" alt="image" src="https://user-images.githubusercontent.com/5015569/169004611-a5e6ac38-49d3-47fd-91ae-0cec3430bdb5.png">

I assume this delay is seen because the conductor worker is waiting to pick up a task from the queue. 

Are there any parameters or configurations we can tweak/increase to reduce the delay and maximize the throughput of workflow execution?

Thank you. 

Kind regards,
Murali

**Comment by apanicker-nflx** [comment]
> this delay is seen because the conductor worker is waiting to pick up a task from the queue.

This observation is correct. The conductor system task workers are waiting to pick up system tasks (like HTTP) from the task queues, since this is an asynchronous operation, you see this delay.

These are the params that can be used to tweak the delay - [Properties](https://github.com/Netflix/conductor/blob/02cfb1c3f40ca31d13640dbb58b4126e59568a3f/core/src/main/java/com/netflix/conductor/core/config/ConductorProperties.java#L87-L113)
